include/nighthawk/common/factories.h

```diff
@@ -33,12 +33,21 @@ class StatisticFactory {
   virtual StatisticPtr create() const PURE;
 };
 
+class RequestSourceConstructorInterface {
```

Comment:
Instead of create another interface, impl and mock for RequestSourceConstructor, would it be cleaner to modify current RequestSourceFactory to create both static and remote RequestSource?

Reply:
Part of the intention of creating the separate interface is to make it easier to test, by allowing us to mock that portion of the code separately. As well as to separate out logic for creating the types of RequestSource so that I can add the RequestSourcePlugin to the RequestSourceConstructor in isolation.